src/components/Dropdown.tsx

```diff
@@ -0,0 +1,109 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+/* eslint-disable jsx-a11y/no-static-element-interactions */
+/* eslint-disable jsx-a11y/anchor-is-valid */
+import React, { useState, useCallback } from 'react';
+import cn from 'classnames';
+import debounce from 'lodash.debounce';
+import { Person } from '../types/Person';
+
+type Props = {
+  peopleFromServer: Person[];
+  onSelected: (person: Person) => void;
+};
+
+export const Dropdown: React.FC<Props> = ({
+  peopleFromServer,
+  onSelected = () => {},
+}) => {
+  const [isActive, setIsActive] = useState(false);
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+
+  const applyQuery = useCallback(
+    debounce((value) => {
+      setAppliedQuery(value.trim() !== '' ? value : '');
```

Comment:
Tbh I didn't get the logic) If you want to trim your string you could do like so
```suggestion
      setAppliedQuery(value.trim());
```
It will be the same